thriftbp/circuit_breaker_test.go

```diff
@@ -0,0 +1,122 @@
+package thriftbp_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/sony/gobreaker"
+
+	"github.com/reddit/baseplate.go/thriftbp"
+	"github.com/reddit/baseplate.go/thriftbp/thrifttest"
+)
+
+var (
+	testMinRequests      = 3
+	testFailureThreshold = .5
+	testMethod           = "method"
+)
+
+func TestShouldTrip(t *testing.T) {
+	cb := newTestCircuitBreaker()
+	counts := gobreaker.Counts{
+		Requests:      uint32(testMinRequests + 1),
+		TotalFailures: 0,
+	}
+	tripped := cb.ShouldTripCircuitBreaker(counts)
+	if tripped {
+		t.Errorf("expected circuit breaker not to trip but it did")
+	}
+}
+
+func TestShouldTrip_zeroRequests(t *testing.T) {
+	cb := newTestCircuitBreaker()
+	counts := gobreaker.Counts{
+		Requests:      0,
+		TotalFailures: 0,
+	}
+	tripped := cb.ShouldTripCircuitBreaker(counts)
+	if !tripped {
+		t.Errorf("expected circuit breaker not to trip but it did")
```

Comment:
There's not any formatting variables in the string so `t.Error` is sufficient, but you will probably need `Errorf` once you converted it into table test.

Reply:
refactored this whole part